Java/Mazes/src/Wall.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Helper class to describe a wall
+ */
+class Wall {
+    // The Wall characteristics
+    int xCoor, yCoor; // Location
+    boolean horz; // Orientation
+
+    /**
+     * Create a wall object
+     * @param xcoord the x-coordinate for the wall
+     * @param ycoord the y-coordinate for the wall
+     * @param horizontal true, if the wall extends to the right of the point, false if it extends down
+     */
+    Wall(int xcoord, int ycoord, boolean horizontal) {
+        if(xcoord < 0 || ycoord < 0) {
```

Comment:
NIT: You can change these into unsigned integers, that way you won't have to do this check.